utils/s3/retryable_http_client.cc

```diff
@@ -76,45 +34,55 @@ future<> retryable_http_client::do_retryable_request(http::request req, http::ex
     }
     uint32_t retries = 0;
     std::exception_ptr e;
-    aws::aws_exception request_ex{aws::aws_error{aws::aws_error_type::OK, aws::retryable::yes}};
+    aws_exception request_ex{aws_error{aws_error_type::OK, retryable::yes}};
     while (true) {
         try {
             e = {};
             co_return co_await (as ? http.make_request(req, handler, *as, std::nullopt) : http.make_request(req, handler, std::nullopt));
-        } catch (const aws::aws_exception& ex) {
+        } catch (const aws_exception& ex) {
             e = std::current_exception();
             request_ex = ex;
         }
 
         if (!_retry_strategy.should_retry(request_ex.error(), retries)) {
             break;
         }
-        co_await seastar::sleep(_retry_strategy.delay_before_retry(request_ex.error(), retries));
+        co_await sleep(_retry_strategy.delay_before_retry(request_ex.error(), retries));
         ++retries;
     }
 
-    if (e) {
-        throw map_s3_client_exception(e);
+    if (_error_handler && e) {
```

Comment:
What if `(bool)_error_handler` is `false`? The exception is ignored? Is that expected?

Reply:
> What if `(bool)_error_handler` is `false`? The exception is ignored? Is that expected?

It is, my assumption if the caller passes empty handler then the caller is not interested in getting the exception.